### PR TITLE
rbd:If the image has any protected snaps,'rbd snap purge image' will return failure without removing any snaps.

### PR DESCRIPTION
**Pull request by mingyuez**
1、Check whether the image has snap or not,if not,do not show any information instead of "Removing all snapshots: 100% complete...done. "
In rbd.cc,function do_purge_snaps
test fix:
An image doesn't have any snap, execute 'rbd snap purge image'
before:
Removing all snapshots: 100% complete...done. 
after:
Without any hints.
2、"rbd snap purge image" will return failure directly without removing any snaps if the image has any protected snaps.
In rbd.cc,function do_purge_snaps
test fix:
An image has 3 snaps(such as a、b、c),and b is a protected snap,then execute 'rbd snap purge image’
before：
Removing all snapshots: 33% complete...failed. and image still have b、c snap.
after：
Removing all snapshots: 0% complete...failed. 
b is a protected snap.

Signed-off-by: mingyuez <zhao.mingyue@h3c.com>

### CODE REVIEW

File: src/rbd.cc [hunk 1]
```diff
@@ -771,12 +771,30 @@ static int do_purge_snaps(librbd::Image& image)
...
+  bool is_protected = false;
   int r = image.snap_list(snaps);
   if (r < 0) {
     pc.fail();
     return r;
   }
-
+   
+  if (0 == snaps.size())
+  {
```
Comment (yuyuyu101) [inline src/rbd.cc]: plz follow the "if-else" brace style

File: src/rbd.cc [hunk 2]
```diff
@@ -771,12 +771,30 @@ static int do_purge_snaps(librbd::Image& image)
...
+	return 0;
+  }
+  for (size_t i = 0; i < snaps.size(); ++i) {
+	r = image.snap_is_protected(snaps[i].name.c_str(), &is_protected);
+	 if (r < 0) {
+		pc.fail();
+		return r;
+	 }
+	 if (is_protected == true)
+	  {
```
Comment (yuyuyu101) [inline src/rbd.cc]: ditto

**Comment by mingyuez** [comment]
@yuyuyu101 I have modified as you said, is it ok?

**Comment by dachary** [comment]
Please ignore the second bot failure : it is a jenkins bug that fails jobs when a rebase happens while it is running.

**Comment by mingyuez** [comment]
@yuyuyu101 I have modified the code as you said, is it ok now?
Looking forward to your reply~

File: src/rbd.cc [hunk 3]
```diff
@@ -771,23 +771,44 @@ static int do_purge_snaps(librbd::Image& image)
...
   int r = image.snap_list(snaps);
   if (r < 0) {
     pc.fail();
     return r;
   }
+   
+  else if (0 == snaps.size())
+  {
+	return 0;
+  }
```
Comment (yuyuyu101) [inline src/rbd.cc]: sorry for my verbose, the style should be:
if (r < 0) {
     pc.fail();
     return r;
   } else if (0 == snaps.size()) {
+	return 0;
+  } else {
+ .....

File: src/rbd.cc [hunk 4]
```diff
@@ -771,23 +771,44 @@ static int do_purge_snaps(librbd::Image& image)
...
+  else {
+	  for (size_t i = 0; i < snaps.size(); ++i) {
+		r = image.snap_is_protected(snaps[i].name.c_str(), &is_protected);
+		 if (r < 0) {
+			pc.fail();
+			return r;
+		 }
+		 else if (is_protected == true)
+		  {
```
Comment (yuyuyu101) [inline src/rbd.cc]: ditto

**Comment by jdurgin** [comment]
The code looks good, just need to fix the style. @yuyuyu101 pointed out the brace style - the indentation also doesn't match. It should be 2 spaces with runs of 8 compressed into tabs (https://github.com/ceph/ceph/blob/master/CodingStyle#L75). vim or emacs will pick up the right indentation settings from the comments at the top of the file.

**Comment by mingyuez** [comment]
@yuyuyu101 I have modified the code again, is the code ok now?

**Comment by yuyuyu101** [comment]
LGTM

**Comment by mingyuez** [comment]
@yuyuyu101  Can you help me merge the commit ?thanks a lot~~

**Comment by mingyuez** [comment]
@yuyuyu101  Thank you very much for your patience!

**Comment by dachary** [comment]
please ignore the bot failure on ubuntu, it was because of a problem that was fixed earlier today

**Pull request merged**